ocs_ci/deployment/deployment.py

```diff
@@ -117,6 +117,27 @@ def label_and_taint_nodes(self):
         ]
         if not worker_nodes:
             raise UnavailableResourceException("No worker node found!")
+        az_worker_nodes = {}
+        for node in worker_nodes:
+            az = node['metadata']['labels'].get(
+                'failure-domain.beta.kubernetes.io/zone'
+            )
+            az_node_list = az_worker_nodes.get(az, [])
+            az_node_list.append(node)
+            az_worker_nodes[az] = az_node_list
+        logger.info(f"Found worker nodes in AZ: {az_worker_nodes}")
+        distributed_worker_nodes = []
+        while az_worker_nodes:
```

Comment:
I am trying to understand the logic here, Is it trying to del node until it comes up in same zone?

Reply:
I am iterate over the AZ, filling the dictionary az_worker_nodes and adding nodes to specific key named by AZ.  Then I have dict like:
```python
az_worker_nodes = {
    'us-east-2a': [node1, node4, node7],
    'us-east-2b': [node2, node5, node8],
    'us-east-2c': [node3, node6, node9],
}
```

Then I am iterate till I have some data in the dictionary.
If there are still keys I am taking one node from each AZ and deleting the node from that list and order appending to this list: distributed_worker_nodes and removing the node from that list.
If the list is already empty under that key, I am deleting that key so this will make sure that the dictionary will get empty once all nodes are taken.

At the end I have list distributed_worker_nodes like:
[node1, node2, node3, node4, node5, node5, node7, node8, node9]
where I know that node are ordered by different AZs like
['us-east-2a', 'us-east-2b', 'us-east-2c', 'us-east-2a', 'us-east-2b', 'us-east-2c', 'us-east-2a', 'us-east-2b', 'us-east-2c']

So when I take nodes 1-6 from that list I will be sure that I have 2 nodes distributed proportionally in all AZs I have.